tests/func/test_init.py

```diff
@@ -64,3 +68,37 @@ def test_cli(self):
         self.assertEqual(ret, 0)
 
         self._test_init()
+
+
+class TestInitQuietShouldNotDisplayWelcomeScreen(TestGit):
+    welcome_msg_1 = boxify(
+        "DVC has enabled anonymous aggregate usage analytics.\n"
+        "Read the analytics documentation (and how to opt-out) here:\n"
+        "{blue}https://dvc.org/doc/user-guide/analytics{nc}".format(
+            blue=colorama.Fore.BLUE, nc=colorama.Fore.RESET
+        ),
+        border_color="red",
+    )
+    welcome_msg_2 = msg = (
+        "{yellow}What's next?{nc}\n"
+        "{yellow}------------{nc}\n"
+        "- Check out the documentation: {blue}https://dvc.org/doc{nc}\n"
+        "- Get help and share ideas: {blue}https://dvc.org/chat{nc}\n"
+        "- Star us on GitHub: {blue}https://github.com/iterative/dvc{nc}"
+    ).format(
+        yellow=colorama.Fore.YELLOW,
+        blue=colorama.Fore.BLUE,
+        nc=colorama.Fore.RESET,
+    )
+
+    @pytest.fixture(autouse=True)
+    def inject_fixtures(self, caplog):
+        self._caplog = caplog
+
+    def test(self):
+        with self._caplog.at_level(logging.INFO, logger="dvc"):
+            ret = main(["init", "--quiet"])
+
+        assert 0 == ret
+        assert self.welcome_msg_1 not in self._caplog.text
+        assert self.welcome_msg_2 not in self._caplog.text
```

Comment:
Big text-in-text test is fragile. I suggest testing for key phrase only, like `"What's next" in caplog.text`.

Reply:
**P.S.** the tricky part with this test is that if welcome message changes it won't break this test and if later welcoming message will show up on `--quiet` this also won't break this test. So `not in` test is always fragile. Consider testing on exact message instead: what should be displayed on `dvc init --quiet`?